core/src/main/java/com/flipkart/gjex/core/web/RotationManagementResource.java

```diff
@@ -0,0 +1,58 @@
+package com.flipkart.gjex.core.web;
+
+import com.flipkart.gjex.core.health.ManualRotationManagement;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+/**
+ * Resource for rotation management
+ * @author ajay.jalgaonkar
+ */
+
+@Singleton
+@Path("/rotation_management")
+@Named
+public class RotationManagementResource {
+
+  private ManualRotationManagement manualRotationManagement;
+
+  @Inject
+  public RotationManagementResource(ManualRotationManagement manualRotationManagement) {
+    this.manualRotationManagement = manualRotationManagement;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/oor")
+  public Response oor() {
+    String response = this.manualRotationManagement.makeOor();
+    return Response.status(Response.Status.OK).entity(response).build();
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/bir")
+  public Response bir() {
+    String response = this.manualRotationManagement.makeBir();
+    return Response.status(Response.Status.OK).entity(response).build();
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/status")
+  public Response status() {
+    String response = this.manualRotationManagement.getStatus();
+    if (this.manualRotationManagement.inHealthy()) {
+      return Response.status(Response.Status.OK).entity(response).build();
+    }
+    return Response.status(Response.Status.NOT_FOUND).entity(response).build();
+  }
```

Comment:
Is this really required? Wouldn't the main heathcheck API simply provide the same?

Reply:
removed